addons/portal/views/portal_templates.xml

```diff
@@ -453,6 +453,10 @@
                 <br/>Please contact us directly for that operation.
             </small>
         </div>
+        <div t-attf-class="mb-3 #{error.get('login') and 'o_has_error' or ''} #{not login and 'd-none' or ''} col-xl-12">
```

Comment:
Hi @msh-odoo 
Hi @msh-odoo,  

We have two templates: `portal_my_details` (t1) and `ticket_validation_screen` (t2), both of which include the `portal_my_details_fields` (t3) template using `t-call`. The `t3` template contains a login field.

Both `t1` and `t2` interact with the controller routes `my/account` (r1) and `pos/ticket/validate` (r2) using `GET` and `POST` methods. However, when `r2` serves `t2` (which includes `t3`), the login field in `t3` will not be displayed in the response to a `GET` request from `r2`. This happens because the condition `#{not login and 'd-none' or ''}` ensures the login field is hidden when `login` is not present in the context of `r2`.

In summary:  
- The login field resides in `t3`.  
- Both `t1` and `t2` utilize `t3`.  
- We want only `t1` to display the login field, not `t2`. This is why the condition exists.

plot twist: i am not able to verify all of this, because `pos/ticket/validate` gives errors in master too. we have to discuss it with POS team.

Reply:
@rahg-odoo We can do two improvements here: 1) Do not add field at all instead of adding ```d-none``` class 2) Instead of adding login field from Controller qcontext we can do the same thing from template from where we are calling template ```portal_my_details_fields```, you can call template using ```t-call``` and pass flag variable login with ```t-set``` variable